gm4_chairs/data/gm4_chairs/functions/kill.mcfunction

```diff
@@ -1,9 +1,14 @@
-#run from main and hit
-# @s = @e[type=pig,tag=gm4_chairs] and NOT inside stair blocks
+# Remove chair and return saddle item
+# @s = pig (chair) to be removed
+# at @s positioned ~ ~.5 ~ / ^ ^.9 ^.2
+# run from main / hit
 
 # spawn saddle item
 loot spawn ~ ~ ~ loot gm4_chairs:drop_saddle
 tag @e[type=minecraft:item,distance=..1,limit=1,sort=nearest] add gm4_chairs_ignore
 
+# dismount if sitting on chair
+execute align xyz positioned ~.5 ~ ~.5 as @a[predicate=gm4_chairs:sitting_in_chair,distance=..0.5] at @s align y run tp @s ~ ~1 ~
```

Comment:
```suggestion
execute align xyz positioned ~.5 ~ ~.5 as @a[predicate=gm4_chairs:sitting_in_chair,distance=..0.5] at @s align y run tp @s ~ ~ ~
```
Should be safer as `~ ~1 ~`. The latter could actually be abused or teleport the player into an invalid location.

Reply:
In what way could this be abused? This teleports the player exactly inside the stair block, so if there's a block below the stair block, it would be exactly on top of that block without falling through the ground